src/test/java/com/game/Battleship/controllers/BoardControllerTests.java

```diff
@@ -1,44 +1,68 @@
 package com.game.Battleship.controllers;
-import com.game.Battleship.models.Board;
-import com.game.Battleship.models.Cell;
-import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.*;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
-import  org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.util.Assert;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @ExtendWith(SpringExtension.class)
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
-
+@AutoConfigureMockMvc
 public class BoardControllerTests {
+
     @Autowired
-    private TestRestTemplate template;
+    private MockMvc mockMvc;
 
     @Test
-    public void boardEndpointContainsSizeAttribute(){
-
-        Board response = template.getForObject("/board", Board.class);
-        Assertions.assertEquals(10, response.getSize());
+    public void testGetRequestToBoardEndpoint() throws Exception {
 
+        this.mockMvc.perform(get("/boards"))
+                .andDo(print()).andExpect(status().isOk());
     }
 
     @Test
-    public void boardFieldSizeIsEqualWithBoardSizeToThePowerOf2(){
-
-        Board response = template.getForObject("/board", Board.class);
-        Assertions.assertEquals(100, response.getBoardField().size());
+    public void testPatchRequestToBoardPlaceEndpoint() throws Exception {
```

Comment:
This test and the one below it isn't fully testing the patch request. Rather than just asserting on the status code, could you also assert on the response of the patch request?

See https://www.logicbig.com/tutorials/spring-framework/spring-web-mvc/http-patch-test.html

Reply:
Done